python/cuml/common/cuda.pyx

```diff
@@ -16,35 +16,11 @@
 
 # distutils: language = c++
 
-import functools
-from libcpp.string cimport string
+import raft.common.cuda
 
-cdef extern from "common/nvtx.hpp" namespace "ML":
-
-    void PUSH_RANGE(string msg)
-
-    void POP_RANGE()
-
-
-def nvtx_range_push(msg: str):
-    """Create a NVTX range with name `msg`"""
-    cdef string s = msg.encode("UTF-8")
-    PUSH_RANGE(s.c_str())
-
-
-def nvtx_range_pop():
-    """End a NVTX range"""
-    POP_RANGE()
-
-
-def nvtx_range_wrap(func):
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        nvtx_range_push(func.__name__)
-        result = func(*args, **kwargs)
-        nvtx_range_pop()
-        return result
-    return wrapper
+nvtx_range_push = raft.common.cuda.nvtx_range_push
+nvtx_range_pop = raft.common.cuda.nvtx_range_pop
+nvtx_range_wrap = raft.common.cuda.nvtx_range_wrap
```

Comment:
We already use the [python nvtx package](https://developer.nvidia.com/blog/nvidia-tools-extension-api-nvtx-annotation-tool-for-profiling-code-in-python-and-c-c/) as opposed to custom bindings in the base class to auto decorate things (https://github.com/rapidsai/cuml/blob/0114c67587af4495222cecd246976cdd2a8ee7cb/python/cuml/common/base.pyx#L21), so we need to remove straggler uses of custom bindings (the calls seem to be in RF and one in arima). I think it'd be cleaner to remove them in this PR and not move custom bindings to raft at all, what do you think?

Reply:
Sure, can do. Thanks for pointing to this!